frontend/src/components/tabs/MetricsTab.tsx

```diff
@@ -0,0 +1,194 @@
+  const isLoading = isLoadingArtifactTypes && isLoadingArtifacts;
+  return (
+    <div className={commonCss.page}>
+      <div className={padding(20)}>
+        <div>
+          This step corresponds to execution{' '}
+          <Link
+            className={commonCss.link}
+            to={RoutePageFactory.executionDetails(execution.getId())}
+          >
+            "{ExecutionHelpers.getName(execution)}".
+          </Link>
+        </div>
+        {isLoading && <Banner message='Metrics is loading.' mode='info' />}
+        {errorArtifacts && (
+          <Banner
+            message='Error in retriving metrics information.'
+            mode='error'
+            additionalInfo={errorArtifacts.message}
+          />
+        )}
+        {errorArtifactTypes && (
+          <Banner
+            message='Error in retriving artifact types information.'
+            mode='error'
+            additionalInfo={errorArtifactTypes.message}
+          />
+        )}
+        {!isLoading && !errorArtifacts && !errorArtifactTypes && artifacts && artifactTypes && (
+          <MetricsSwitcher artifacts={artifacts} artifactTypes={artifactTypes}></MetricsSwitcher>
+        )}
+      </div>
+    </div>
+  );
+}
+
+/**
+ * Visualize system metrics based on artifact input.
+ */
+function MetricsSwitcher({ artifacts, artifactTypes }: MetricsSwitcherProps) {
+  // system.ClassificationMetrics contains confusionMatrix or confidenceMetrics.
+  // TODO: Visualize confusionMatrix using system.ClassificationMetrics artifacts.
+  // https://github.com/kubeflow/pipelines/issues/5668
+  let classificationMetricsArtifacts = filterArtifactsByType(
+    'system.ClassificationMetrics',
+    artifactTypes,
+    artifacts,
+  );
+
+  const confidenceMetricsArtifacts = classificationMetricsArtifacts
+    .map(artifact => ({
+      name: artifact
+        .getCustomPropertiesMap()
+        .get('name')
+        ?.getStringValue(),
+      customProperties: artifact.getCustomPropertiesMap(),
+    }))
+    .filter(x => !!x.name)
+    .filter(x => {
+      const confidenceMetrics = x.customProperties
+        .get('confidenceMetrics')
+        ?.getStructValue()
+        ?.toJavaScript();
+      return !!confidenceMetrics;
+    });
+
+  return (
+    <>
+      {(!confidenceMetricsArtifacts || confidenceMetricsArtifacts.length === 0) && (
+        <Banner message='There is no metrics artifact available in this step.' mode='info' />
+      )}
+      {classificationMetricsArtifacts &&
+        classificationMetricsArtifacts.length > 0 &&
+        classificationMetricsArtifacts.map(artifact => {
+          const id = artifact.getId();
+          const customProperties = artifact.getCustomPropertiesMap();
+          const confidenceMetrics = customProperties
+            .get('confidenceMetrics')
+            ?.getStructValue()
+            ?.toJavaScript();
+          return (
+            <React.Fragment key={id}>
```

Comment:
Can be future work:
I suggest wrapping each visualization with an ErrorBoundary, so that if one visualization crashes because of data integrity problems, the entire UI doesn't crash. (we used many type casting, and did not validate user data schema, so it's likely bad user data might crash the entire UI. it's better to only crash the single visualization and show user what data is wrong)

See https://reactjs.org/docs/error-boundaries.html

Reply:
also, `id` is probably not a good key here, because we can see that one artifact might contain both confusionMatrix and roc curve, and they share the same ID.

Reply:
Thank you for the reference! I would love to use error boundaries, but its `getDerivedStateFromError` is not yet supported in react hook https://reactjs.org/docs/hooks-faq.html#do-hooks-cover-all-use-cases-for-classes. I would like to use it when it is available.

I have also updated the key to `metrics type name` + `id`, if this also makes sense to you.

Reply:
Implemented ErrorBoundary based on offline discussion.